gruppeH/dev

olivermjmj opened the pull request:
Dette er GryppeH

olivermjmj committed:
added movieMapper and the ability to get all movies from the db

olivermjmj committed:
Now a user can search for a specefic genre and get all the movies from called genre.

olivermjmj committed:
Er nu færdig med movieController

Commit:
Added controller to FilmRouletten

Commit:
Made a new folder (FilmRouletten) for the HTML, and incorporated the entire html for FilmRouletten in the index.html file

Commit:
Made a logo 200x200 pixels for the FilmRoulette project

Commit:
Made a controller for FilmRouletten

Commit:
Created a new css file for FilmRouletten

Commit:
Fixed minor spelling errors in the HTML

olivermjmj committed:
updated movieController to get a single random movie instead of a selected one.

olivermjmj committed:
Changed name getAMovieByGenre

olivermjmj committed:
Merge branch 'gruppeH/feature_movie' into gruppeH/dev

olivermjmj committed:
merged with movie feature.

olivermjmj committed:
sql name change and fixed movieController and movieMapper

olivermjmj committed:
sql name change and fixed movieController and movieMapper

Commit:
Fixed manglende ID til Roulettehjulet samt knappen til det.

olivermjmj committed:
Made RouletteIconSpin.js which makes the icon spin 180 degrees afterwards it stops and goes toward the left side of the monitor where it also spins another 180 degrees.

Added the animation of it in wheelAnimation.css

Commit:
Gav id til rouletteWheel

olivermjmj committed:
added wheelAnimationSlide.css and logic for it by js.

CarlEmilKoehn committed:
login, create user, test html sider, usermapper og usercontroller (mostly done)

CarlEmilKoehn committed:
reverted main og smidt ændringer i routing ind i usercontroller

olivermjmj committed:
Merge remote-tracking branch 'origin/gruppeH/feature_user' into gruppeH/dev

olivermjmj committed:
update of user.

olivermjmj committed:
changed file structure.

olivermjmj committed:
updated more file structure.

Commit:
Har lavet en struktur til forsiden i css. mangler præcis placering og farve.

Commit:
Merge branch 'gruppeH/dev' into gruppeH/CSS

# Conflicts:
#	src/main/resources/public/css/FilmRouletten.css

Commit:
tilføjede et roulette hjul i bedre kvalitet

Commit:
oprettede en watchlist og en login knap i HTML

Commit:
oprettede login og watchlist knapperne i css

CarlEmilKoehn committed:
provider update

olivermjmj committed:
added login and register to the pathway.

olivermjmj committed:
.

olivermjmj committed:
reroll

olivermjmj committed:
removed js

olivermjmj committed:
1 am

olivermjmj committed:
3 pm

CarlEmilKoehn committed:
Merge remote-tracking branch 'origin/gruppeH/provider' into gruppeH/dev

CarlEmilKoehn committed:
:)

olivermjmj committed:
added functionality for seeing providers.

olivermjmj committed:
added watchlist

CarlEmilKoehn committed:
Fix af watchlist kind of

olivermjmj committed:
Merge remote-tracking branch 'origin/gruppeH/html-fix' into gruppeH/dev

olivermjmj committed:
updated logic

olivermjmj committed:
structure update

olivermjmj committed:
structure update

olivermjmj committed:
structure update

tjorne commented on .idea/misc.xml:
why is this relevant to add to the repo?

tjorne commented on src/main/java/app/Main.java:
```diff
@@ -1,7 +1,10 @@
 package app;
 
 import app.config.ThymeleafConfig;
-import app.controllers.*;
```
why not continue to use the wildcard?

tjorne commented on src/main/java/app/Main.java:
```diff
@@ -25,9 +28,14 @@ public static void main(String[] args)
+        //UserController.addRoutes(app);
+        //TimeZonesController.addRoutes(app);
+
+        app.get("/", ctx -> ctx.render("filmRouletten/index.html"));
```
I believe you just hijacked the entire frontpage of the application by this change. Please revert this.

tjorne commented on src/main/java/app/Main.java:
```diff
@@ -25,9 +28,14 @@ public static void main(String[] args)
+
+        app.get("/", ctx -> ctx.render("filmRouletten/index.html"));
         UserController.addRoutes(app);
-        TimeZonesController.addRoutes(app);
```
Why did you remove the TimeZonesController?

tjorne commented on src/main/java/app/controllers/UserController.java:
Why are these empty lines relevant to add to the codebase?

tjorne commented on src/main/resources/public/css/styles.css:
Please don't overwrite the shared styles.css file! 

It seems you've also created 3 .css files of your own, but they don't seem to contain the same content as what you have replaced styles.css with.

tjorne commented on src/main/resources/public/css/filmRouletten/filmRoulettenGetMovie.css:
```diff
@@ -0,0 +1,313 @@
+    box-sizing: border-box;
+}
+
+:root {
```
This is duplicate of filmRoulettenIndex.css and a lot of the following content seem to be as well. Please consolidate.

tjorne commented on src/main/resources/templates/FilmRouletten/login.html:
why the need to create your own login page? Did you not like the other one?

tjorne commented on src/main/resources/templates/FilmRouletten/createuser.html:
why the need to create your own createuser page? Did you not like the other one?

tjorne commented on src/main/resources/sql/filmRouletten/filmRouletten.sql:
```diff
@@ -0,0 +1,82 @@
+    CONSTRAINT provider_name_roulette_pkey PRIMARY KEY (provider_name_id)Har
+    );
+
+CREATE TABLE IF NOT EXISTS public.user_roulette
```
I don't really see the need for your own user table here, as it seem to be redundant as per the exisiting user table

tjorne commented on src/main/resources/templates/createuser.html:
I admit we didn't go into details regarding the code standard enough to discuss the age-old question regarding 2 vs 4 spaces for tab sizes. 

But if you prefer the 2-space tab size for indentation, I believe it should be an entire Pull Request on its own, and even a meeting where we could discuss it and settle on the standard prior to the PR. 

As for this PR, please revert the changes to this file.

tjorne commented on src/main/resources/templates/fragments.html:
why is this relevant?

tjorne commented on src/main/resources/templates/index.html:
It is difficult to see what you have changed, as it at first glance looks as though you've overwritten the entire class with itself, just with another formatting. 

Did you add your own card here? that was the purpose, but maybe it got lost, as it seems you've also accidentally highjacked the main index page.

tjorne commented on src/main/resources/public/js/ProviderPopup.js:
nice.

tjorne commented on src/main/java/app/persistence/filmRouletten/UserMapper.java:
This is somewhat a duplicate of the existing UserMapper. Yours, however, seems to extend a bit upon it, which is nice. You are more than welcome to create a new PR, improving upon the shared codebase.

tjorne commented on src/main/java/app/persistence/UserMapper.java:
These changes kills the entire application for everybody else. 

Are you proposing every other group use the filmRouletten/UserMapper instead?

tjorne replied:
This still isn't completely resolved. 

It looks as though, you've reverted the changes to this file, however then I don't understand the changes below.

tjorne commented on src/main/java/app/persistence/ConnectionPool.java:
Your provided sql scripts will ensure that whatever you need in the database is added to the existing one, so theres no need to change it to your own. 
The "FilmRouletten" database also doesn't exist on the provided server.

tjorne commented on src/main/java/app/entities/filmRouletten/Provider.java:
good 👍

tjorne commented on src/main/java/app/entities/filmRouletten/Movie.java:
good 👍

tjorne commented on src/main/java/app/entities/User.java:
please do not change the shared User resource as it might delete functionality needed by other groups.

tjorne commented on src/main/java/app/controllers/filmRouletten/FilmRoulettenController.java:
I'm having a hard time figuring out if this is in use. is it?

tjorne reviewed (changes requested):
Jeg glæder mig til at se jeres projekt virke, men jeg har med vilje ikke villet køre jeres sql kode, da den ser ud til at lave en redundant user tabel. 

Der er en del af jeres file changes som berører syntax ændringer eller andre ændringer i den fælles kode base, hvilket vi skal huske at være meget opmærksomme på, da de kan have konsekvenser for andre dele af systemet og derved berøre andre grupper/udviklere af det fælles projekt.

Jeg har efterladt en hel stak kommentarer til mere specifikke steder i jeres kode, hvor der er nogle ting der skal rettes op før det kan merges ind. 

Jeg håber I ser ændringerne som forslag til forbedringer og ikke kritik :) 

I kan altid svare på mine kommentarer hvis I har spørgsmål/kommentarer.

tjorne commented:
> Dette er GryppeH

Jeg savner en beskrivelse af hvad der sker i denne branch og hvorfor den skal ind i master. 

Et lille sæt instrukser til hvordan man kører det og tester det, vil også hjælpe den udvikler der skal review'e dette PR.

saldic committed:
revert to original misc.xml

Overlooked commit error

saldic committed:
Delete src/main/java/app/controllers/filmRouletten/FilmRoulettenController.java

File is deleted, as it was a miscommunication. The replacement of FilmRoulettenController.java will be seen in MovieController.java.

saldic committed:
Delete src/main/java/app/controllers/UserController.java

A new UserController.java was created, and there is no real use for this anymore.

saldic committed:
Update User.java

Reverted back the changes to the original User class. However I did leave  the " public User(int userId, String userName, String password)" constructer in place, as we're using it in our part of the code, And I dont think it will have an effect on other groups.

saldic committed:
Update UserMapper.java

Removed block comments from UserMapper.

Commit:
Fixed User to it's original state, removed random non eligible code blocks.

tjorne commented on src/main/java/app/controllers/UserController.java:
You have deleted the shared userController, which breaks the application for everybody else. 

You shouldn't touch this one in this PR. If you have enhancements for it, please add them in a separate PR. 

For now - please revert this change. I realize this might be conflicting with the filmRouletten/UserController.java that you have added, so that one should be the one to go. 

If you need anything specifically from that one, please qualify it so it is specific to filmRouletten and ensure it works with the other one still.

tjorne commented:
Der er stadig kommentarer der mangler at blive "resolved". Hvis I har udbedret tingene - er I meget velkomne til at trykke "resolve" selv, så vi kan holde styr på hvor langt vi er i processen. 

Der er stadig nogle ting vedr. UserController og eventuelt også stadig UserMapper der ødelægger projektet for alle andre, som skal fikses før vi kan komme videre.

saldic commented:
Roger that, vi troede nemlig deadline for den sidste pr var i fredags.
Er der en deadline for hvornår at vi skal færdige med review?
For jeg tænker helt klart at vi vil få det løst, så vi kan blive erklæret færdige.

Sendt fra Outlook til iOS<https://aka.ms/o0ukef>
________________________________
Fra: Jesper Tjørnelund ***@***.***>
Sendt: Sunday, October 12, 2025 11:53:21 AM
Til: tjorne/lifehackE25 ***@***.***>
Cc: Dino Saldic ***@***.***>; Assign ***@***.***>
Emne: Re: [tjorne/lifehackE25] gruppeH/dev (PR #10)

Du får ikke ofte mails fra ***@***.*** Få mere at vide om, hvorfor dette er vigtigt<https://aka.ms/LearnAboutSenderIdentification>

@tjorne requested changes on this pull request.

________________________________

On src/main/java/app/controllers/UserController.java<https://github.com/tjorne/lifehackE25/pull/10#discussion_r2423456994>:

You have deleted the shared userController, which breaks the application for everybody else.

You shouldn't touch this one in this PR. If you have enhancements for it, please add them in a separate PR.

For now - please revert this change. I realize this might be conflicting with the filmRouletten/UserController.java that you have added, so that one should be the one to go.

If you need anything specifically from that one, please qualify it so it is specific to filmRouletten and ensure it works with the other one still.

________________________________

On src/main/java/app/persistence/UserMapper.java<https://github.com/tjorne/lifehackE25/pull/10#discussion_r2423458354>:

This still isn't completely resolved.

It looks as though, you've reverted the changes to this file, however then I don't understand the changes below.

—
Reply to this email directly, view it on GitHub<https://github.com/tjorne/lifehackE25/pull/10#pullrequestreview-3328026543>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BPF3HHGAFNJNESTPGNMBFYL3XIQJDAVCNFSM6AAAAACIW4OBVGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTGMRYGAZDMNJUGM>.
You are receiving this because you were assigned.Message ID: ***@***.***>
